core/src/main/scala/org/bitcoins/core/wallet/signer/Signer.scala

```diff
@@ -243,6 +246,46 @@ sealed trait BitcoinSignerSingle[-SpendingInfo <: BitcoinUTXOSpendingInfoSingle]
 
 object BitcoinSignerSingle {
 
+  def sign(
+      psbt: PSBT,
+      inputIndex: Int,
+      signer: Sign,
+      conditionalPath: ConditionalPath = ConditionalPath.NoConditionsLeft,
+      isDummySignature: Boolean)(
+      implicit ec: ExecutionContext): Future[PSBT] = {
+    // if already signed by this signer
+    if (psbt
+          .inputMaps(inputIndex)
+          .getRecords[PartialSignature](PartialSignatureKeyId)
+          .exists(_.pubKey == signer.publicKey)) {
+      throw new IllegalArgumentException(
+        "Input has already been signed with this key")
+    }
+
+    val tx = psbt.transaction
+    val spendingInfo =
+      psbt
+        .inputMaps(inputIndex)
+        .toUTXOSpendingInfoSingle(tx.inputs(inputIndex),
+                                  signer,
+                                  conditionalPath)
+
+    val sigAndPubKeyF =
+      signSingle(spendingInfo, tx, isDummySignature)
+
+    sigAndPubKeyF.map { sigAndPubKey =>
+      val pubKey = sigAndPubKey._1
+      val signature = sigAndPubKey._2
+
+      val map = psbt.inputMaps(inputIndex)
+      val newElements = map.elements :+ PartialSignature(pubKey, signature)
+
+      val newInputMaps =
+        psbt.inputMaps.updated(inputIndex, InputPSBTMap(newElements))
```

Comment:
It would be nice if `PSBT` had more methods available for doing stuff like this so that users of `PSBT` don't have to go updating the guts of `Vector`s. For example in this case it would be nice to have a `PSBT.addSignature(index: Int, pubKey: ECPublicKey, sig: ECDigitalSignature)`